orca-mine/src/main/groovy/com/netflix/spinnaker/orca/mine/tasks/MonitorCanaryTask.groovy

```diff
@@ -35,6 +35,8 @@ class MonitorCanaryTask extends AbstractCloudProviderAwareTask implements Retrya
   long backoffPeriod = TimeUnit.MINUTES.toMillis(1)
   long timeout = TimeUnit.DAYS.toMillis(2)
```

Comment:
This property overrides a `getTimeout` method in `Task`. I think it might make sense to just have it compute a value rather than returning something fixed.

There's some upcoming work that will move timeout from the task to the stage level which will make a lot more sense.

Reply:
I would do that, but I don't see any way to access the context of the stage (where the canary configuration is) from the task itself.

Reply:
yeah, that's not ideal is it

Reply:
Heh, no. We could also just bump the timeout to 72 hours for now and hope nobody goes over it (seems like it took two years for someone to configure a 48-hour canary) until we can do smarter things with stage-level timeouts.

Reply:
I think given @emjburns is going to be tackling some rationalization of timeout behavior we should merge this & revisit it as part of the move from task-level timeouts to stage level.